### ISSUE TITLE
find(root, key) call in erase method seems to be unimplemented

### ISSUE THREAD

**Issue by dbiton** [NicoG60/interval-tree#3]
Hey, first of all - great job! this is some really impressive work.
I can't get the code to compile due to the "node* n = find(root, key);" call in the erase method.

specifically, getting: 

'interval_tree<uint64_t,SpltSketch::Node *,std::less<uint64_t>,0>::find': no overloaded function takes 2 arguments', code C2661 in line 531.

looked around for a similarly named function but I don't want to replace the call with a wrong one and mess something up.

**Comment by NicoG60** [comment]
Hi,
Thanks for the kind word! I'm happy to give a hand debugging that but I'll need a bit more context.
Obviously I wont be able to scan through your entire code base but if you could make a minimal example that reproduce that bug that'd be great.

**Comment by NicoG60** [comment]
Also, If you feel like you know what you're doing, feel free to make a PR.

**Comment by NicoG60** [comment]
Ahoi again
I had a quick look and indeed, that function doesn't exists. It's a missing case in the tests. Could you try with the PR I created? If that works as expected, I'll add the test case and merge that.

**Comment by dbiton** [comment]
Windows 10
Visual Studio 19

```
#include "interval_tree.h"

int main() {
	interval_tree<uint64_t, uint64_t> tree;
	return 0;
}
```

getting 'is_nothrow_swappable is not a member of std', switched from c++14 to c++17. getting "could not deduce template" for the emplace in lines 428, 433.

messed around a bit and it seems to work with int while uint64_t causes problems.

**Comment by NicoG60** [comment]
Hey Again,
I'm afraid I wont be able to help immediately here. I'm not using Windows and I need to spin up a VM and stuff.
If you're up for it and can find your way through the codebase, let me know and make a PR.
If you don't want to do that, let me know as well and I'll have a look later this week.

**Comment by dbiton** [comment]
I'll try to make a PR - I don't have much experience with GitHub but I'm happy to give it a shot.
I appreciate your help!

**Comment by NicoG60** [comment]
Very simple to make a PR:
- Fork the repo on your own account
- Make your change, commit and push them
- go back on GitHub, in Your fork the option should appear like "Hey, you made some changes -> push a PR"

**Comment by NicoG60** [comment]
Fixed by #5 it seem! Thanks again
